improver_tests/wxcode/wxcode/test_WeatherSymbols.py

```diff
@@ -327,6 +327,25 @@ def test_incorrect_units(self):
         with self.assertRaisesRegex(ValueError, msg):
             self.plugin.check_input_cubes(self.cubes)
 
+    def test_returns_used_cubes(self):
+        """Test that check_input_cubes method returns a list of cubes that is
+        reduced to include only those diagnostics and thresholds that are used
+        in the decision tree.
+
+        The test below is that the 0.1 mm/hr sleet rate threshold (unused) is
+        not contained in the returned cube list, whilst the 1 mm/hr is."""
+
+        result, _ = self.plugin.check_input_cubes(self.cubes)
+
+        expected = iris.Constraint(
+            coord_values={"lwe_sleetfall_rate": lambda cell: 2.7e-07 < cell < 2.8e-07}
+        )
+        unexpected = iris.Constraint(
+            coord_values={"lwe_sleetfall_rate": lambda cell: 2.7e-08 < cell < 2.8e-08}
+        )
+        self.assertTrue(len(result.extract(expected)) > 0)
+        self.assertEqual(len(result.extract(unexpected)), 0)
```

Comment:
Should we perhaps check more thorough check of what is returned as 'result'? (i.e. other than 'lwe_sleetfall_rate')

Reply:
I've modified this test to check all the expected diagnostic/threshold pairings are present, and that the ones which should be omitted are missing.